ui/src/app/settings/views/Scripts/ScriptsUpload.js

```diff
@@ -20,8 +20,8 @@ const ScriptsUpload = ({ type }) => {
   const hasErrors = useSelector(scriptSelectors.hasErrors);
   const errors = useSelector(scriptSelectors.errors);
   const saved = useSelector(scriptSelectors.saved);
-  const [savedScript, setSavedScript] = useState();
-  const [script, setScript] = useState();
+  const [savedScript, setSavedScript] = React.useState();
+  const [script, setScript] = React.useState();
```

Comment:
I wasn't able to find a way to easily mock a named import, hence this change here. I'm usually loathe to make changes to code to facilitate tests, but in this case I couldn't find a solution - suggestions welcome.

Reply:
#### ScriptUpload.test.js
``` js
import * as React from "react";
```
Seems to work and then you don't need to use `React.useState();` in `ScriptUpload.js`.

Reply:
Nice! That's a much better solution, thank you @Caleb-Ellis!

Reply:
@Caleb-Ellis hmm, I can't actually get that to work, do you have the hasMetadata test passing?

Reply:
@Caleb-Ellis based on @hatched's suggestion, I refactored the file reading code out, and mocked it for these tests, which meant I can make assertions against store.getActions() which is much nicer I think.

Reply:
Agreed, this is a much more sensible separation!